### PR TITLE
Feature/png 21

### PR DESCRIPTION
**Pull request by mbuehrig**

### CODE REVIEW

File: config/scaffold/remove.js [hunk 1]
```diff
@@ -12,36 +15,57 @@ var nodeFiles = [];
...
+        var pinguJsFile = fs.readFileSync(pahtsHelper.pinguAppJs);
+
+        var jsResult = pinguJsFile.replace(
+          new RegExp(`import ${node.component} .*${nodeType.toLowerCase()}s\/${node.directory}\/${node.file}';\n`, 'g'),
+          ''
+        );
+
+        fs.writeFile(pathsHelper.pinguAppJs, jsResult);
+      }
+    }));
```
Comment (tschuems) [inline config/scaffold/remove.js]: remove semicolon, else the task won't start

**Review by tschuems — COMMENTED** [review]
Looks good, but one feature is missing: 

If you import an component with a js, it should apply the import to pinguapp AND directly use the import in the pinguapp constructor.

in example:

```
import TestComponent from './components/TestComponent/testComponent';

class PinguApp {
  constructor() {
    this.components = {};
    this.components.testComponent = TestComponent;
  }
}

```

this should autoinsert this aswell.

File: config/scaffold/remove.js [hunk 2]
```diff
@@ -12,36 +15,57 @@ var nodeFiles = [];
...
+      if (hasSCSS) {
+        var scssFile = fs.readFileSync(pathsHelper.mainScss, 'utf8');
+
+        var scssResult = scssFile.replace(new RegExp(`@import .*${nodeType.toLowerCase()}s\/${node.directory}\/${node.file}';\n`, 'g'), '');
+
+        fs.writeFile(pathsHelper.mainScss, scssResult);
+      }
+
+      if (hasJS) {
+        var pinguJsFile = fs.readFileSync(pahtsHelper.pinguAppJs);
```
Comment (tschuems) [inline config/scaffold/remove.js]: error -> `pahtsHelper` doesn't exists 
its `pathsHelper`

**Comment by mbuehrig** [comment]
No. We said we want to load the Components by an autohandler so only the components are initialized which are on the page.

File: config/scaffold/add.js [hunk 3]
```diff
@@ -40,6 +67,50 @@ gulp.task('addGenerator', ['addGeneratorPrompts'], function() {
...
+          var fileRead = fs.readFileSync(pathsHelper.pinguAppJs, 'utf8');
+
+          var result = fileRead.replace(scaffoldUtils.JS_AUTOIMPORT_REGEX, [
+            `import ${nodeName.component} from '../../../${folder}s/${nodeName.directory}/${nodeName.file}';`,
+            '\n',
+            '// autoimportcomponent',
+          ].join(''));
+
+          var result2 = result.replace(scaffoldUtils.JS_COMPONENT_REGEX, [
+            `this.components.${nodeName.component} = ${nodeName.component}`,
```
Comment (tschuems) [inline config/scaffold/add.js]: semicolon is missing at the end

**Review by tschuems — APPROVED** [review]

**Pull request merged**